admin-dev/themes/new-theme/js/pages/product/components/specific-price/specific-price-list-renderer.ts

```diff
@@ -102,23 +105,23 @@ export default class SpecificPriceListRenderer implements RendererType {
       editBtn.dataset.specificPriceId = String(specificPrice.id);
 
       if (!specificPrice.period) {
-        periodField.textContent = String(periodField.dataset.unlimitedText);
+        periodField.textContent = String(periodField?.dataset.unlimitedText);
       } else {
         periodFromField.textContent = specificPrice.period.from;
         periodToField.textContent = specificPrice.period.to;
       }
 
-      tbody.append(trClone);
+      tbody?.append(trClone);
       this.addEventListenerForDeleteBtn(deleteBtn);
     });
   }
 
   private toggleListVisibility(show: boolean): void {
-    this.listContainer.classList.toggle('d-none', !show);
+    this.listContainer?.classList.toggle('d-none', !show);
   }
 
-  private selectListField(templateTrClone: HTMLElement, selector: string): HTMLElement {
-    return templateTrClone.querySelector(selector) as HTMLElement;
+  private selectListField(templateTrClone: HTMLElement, selector: string): HTMLElement | null | undefined {
+    return templateTrClone.querySelector<HTMLElement>(selector);
```

Comment:
@NeOMakinG 
Sorry bother you, but I have several doubts about this change (line 123 and 124), because at line #79 the code start to using this function to assign an element to the several const. After that we try to assign a value to the textContent param (of each const)
Obviously I have a lot of warnings from my IDE because the return type could be null or undefined.
What to do? Check every single const if is not null or undefined?